client/src/components/TodoWrapper.js

```diff
@@ -4,23 +4,34 @@ import { TodoForm } from './TodoForm'
 import { Todo } from './Todo'
 import {v4 as uuidv4} from 'uuid'
 import { EditTodoForm } from './EditTodoForm';
+
 uuidv4();
 
 export const TodoWrapper = () => {
     const [todos, setTodos] = useState([]);
+    const [error, setError] = useState(null);
+    const [loading, setLoading] = useState(false);
+
     useEffect(() => {
-        axios.get('http://localhost:8055/items/todolist')
-            .then((response) => {
-                setTodos(response.data.data);
-            })
-            .catch((error) => {
-                console.error("Error fetching todos:", error);
-            });
+        fetchTodo();
     }, []);
 
+    const fetchTodo = async () =>{
+        setLoading(true);
+        try {
+            const response = await axios.get('http://localhost:8055/items/todolist');
+            setTodos(response.data.data);
+        } catch (error) {
+            setError("Error fetching todos");
+            console.error("Error fetching todos:", error);
+        } finally {
+            setLoading(false);
+        }
+    }
+
```

Comment:
make seperate utils function 

step to reproduce : 
1. create folder called `/utils`
2. make a file called `/axios-adapter.js`
3. inside of it you can create an axios intercetor,for refference go here : https://axios-http.com/docs/interceptors

example of axios interceptor in minisoccer code, you can go here : https://github.com/disketsapm/minisoccer-web-client/blob/main/services/request-adapter.service.ts

**you will be invited to the minisoccer client repo**